### PR TITLE
Allow optional charset in source maps

### PR DESCRIPTION
**Pull request by mkhl**
Closes #1

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -15,6 +15,11 @@
   "bugs": {
     "url": "https://github.com/faucet-pipeline/extract-sourcemap/issues"
   },
-  "author": "Lucas Dohmen <lucas@dohmen.io> (https://lucas.dohmen.io/)",
-  "license": "MPL-2.0"
+  "license": "MPL-2.0",
+  "devDependencies": {
+    "mocha": "^5.2.0"
+  },
+  "scripts": {
```
Comment (FND) [inline package.json]: FWIW, I'd put `scripts` abelow `bin` and `license` below `author`.

File: index.js [hunk 2]
```diff
@@ -1,12 +1,22 @@
...
+let findUrl = /sourceMappingURL=data:(\S+)/;
+let findMap = /;base64,(\S+)$/;
+
+function firstMatchedGroup(regex, data) {
+	let matches = regex.exec(data);
+	if(!matches) {
+		return null;
+	}
+	return matches[1];
+}
```
Comment (FND) [inline index.js]: FWIW, I like to put utility functions towards the bottom, reserving the top of a module for more significant bits. ( ❤️ function hoisting)

**Review by FND — COMMENTED** [review]
lgtm 👍

If it was me, I'd just squash all of this down to two commits, but luckily I don't have to concern myself with such things here.

**Comment by moonglum** [comment]
Thanks for pairing on this, @mkhl 🙇 I adjusted the parts we discussed as well as @FND's feedback and some additional improvements like a README.

**Pull request merged**